src/videojs-hls.js

```diff
@@ -307,40 +307,30 @@ videojs.Hls.prototype.setupSourceBuffer_ = function() {
     var
       segmentInfo = this.pendingSegment_,
       segment,
+      segments,
       playlist,
       currentMediaIndex,
       currentBuffered,
       timelineUpdates;
 
-    // stop here if the update errored or was aborted
-    if (!segmentInfo) {
```

Comment:
This was being called twice. The correct place to call it is after `pendingSegment_` is reset.